src/main/java/seedu/address/model/attendance/AttendancePresence.java

```diff
@@ -0,0 +1,72 @@
+package seedu.address.model.attendance;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+/**
+ * Represents the presence of a student in an attendance record.
+ * It can either be 'present' or 'absent'.
+ */
+public class AttendancePresence {
+
+    public static final String MESSAGE_CONSTRAINTS =
+            "Presence field should only contain either 'present' or 'absent', and it should not be blank";
+
+    public final String attendancePresence;
+
+    /**
+     * Constructs an {@code AttendancePresence} with the specified presence.
+     *
+     * @param presence A valid presence, either 'present' or 'absent'.
+     */
+    public AttendancePresence(String presence) {
+        requireNonNull(presence);
+        presence = presence.toLowerCase(); // Normalize to lowercase for case-insensitive comparison.
+        checkArgument(isValidInput(presence), MESSAGE_CONSTRAINTS);
+        attendancePresence = presence;
+    }
+
+    /**
+     * Returns true if the given string is a valid presence, which can be either 'present' or 'absent'.
+     *
+     * @param test The string to check for validity.
+     * @return True if the string is a valid presence, false otherwise.
+     */
+    public static boolean isValidInput(String test) {
+        return test.equals("present") || test.equals("absent");
```

Comment:
Might want to use regex for this.
`public static boolean isValidInput(String test) {
    return test.matches("(?i)^(present|absent)$");
}`

^ not sure if correct haha :P

Reply:
you are right, perhaps not regex, but rather use a static final variable to test equality